CoreFoundation/Base.subproj/SwiftRuntime/TargetConditionals.h

```diff
@@ -76,21 +77,31 @@
 #define TARGET_OS_LINUX        0
 #define TARGET_OS_WINDOWS      0
 #define TARGET_OS_BSD          0
+#define TARGET_OS_CYGWIN       0
 #elif __linux__
 #define TARGET_OS_DARWIN       0
 #define TARGET_OS_LINUX        1
 #define TARGET_OS_WINDOWS      0
 #define TARGET_OS_BSD          0
+#define TARGET_OS_CYGWIN       0
+#elif __CYGWIN__
+#define TARGET_OS_DARWIN       0
+#define TARGET_OS_LINUX        0
```

Comment:
Cygwin provides the POSIX system calls and environment, and is similar to Linux because Linux also confirms to POSIX. But some functions like ```eventfd```, ```epoll```, ```timerfd_create```  seems to be never implemented in Cygwin because they are not POSIX standard.
Anyway, I agree your method will simplify this patch and I will modify if others has no opposing opinion, although there was similar opposing discussions for compiler. [swift #2351](https://github.com/apple/swift/pull/2351#discussion_r61667167)